packages/p2p/src/components/app.jsx

```diff
@@ -74,14 +127,17 @@ class App extends Component {
                 updated_orders[idx_order_to_update] = order_response;
             }
             // trigger re-rendering by setting orders again
-            this.setState({ orders: updated_orders });
+            this.setState({ orders: updated_orders }, () => this.handleNotifications(updated_orders));
```

Comment:
why does this need to be after state update and not the next line?

Reply:
setstate is an `async` call, second parameter of setState is a callback after assigning the state value

Reply:
I understand, I'm asking why it's needed when in the above if condition it's not needed

Reply:
you're right, ill make it the same